### PR TITLE
ds_prefill(Add TtParallelEmbedding)

### PR DESCRIPTION
**Pull request by mbezuljTT**
### Ticket
Related to DeepSeek V3 prefill implementation (#2603)

### Problem description
The DeepSeek V3 prefill pipeline needs a parallel embedding layer that converts token IDs into hidden representations distributed across a 2D mesh (SP + TP).

### What's changed
Add `TtParallelEmbedding` module and PCC test:

- **`tt_parallel_embedding.py`**: `LightweightModule` that shards embedding weight on `emb_dim` across TP axis (mesh cols) and replicates across SP axis (mesh rows). Each device does an independent lookup on its weight slice — no CCL needed. Output: `[1, 1, isl_per_chip, emb_dim/tp_factor]` TILE_LAYOUT.
- **`test_parallel_embedding.py`**: PCC test comparing against `torch.nn.functional.embedding` with small and production-scale (vocab=129280, emb=7168) configs on linear-4 and mesh-2x4 topologies. Threshold: 0.999. Signpost label and all internal variables use `isl_per_chip` to identify per-chip sequence length.

### Checklist

- [x] <a href="https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-e2e-tests.yaml?query=branch:mbezulj/2603-tok-emb-layer"> </a>
- [x] <a href="https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-e2e-tests.yaml?query=branch:mbezulj/2603-tok-emb-layer"> </a>
- [x] <a href="https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-deepseek-prefill-tests.yaml?query=branch:mbezulj/2603-tok-emb-layer"> </a>
- [x] <a href="https://github.com/tenstorrent/tt-metal/actions/workflows/demo-sp-release.yaml?query=branch:mbezulj/2603-tok-emb-layer"> </a> ⚠️ waiver: wan failing irrelevant;

### CODE REVIEW

File: models/demos/deepseek_v3_d_p/tests/pcc/test_parallel_embedding.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+)
+def test_parallel_embedding(mesh_device, isl_per_chip, vocab_size, emb_dim):
+    """Test TtParallelEmbedding against torch reference."""
+
+    sp_factor = mesh_device.shape[0]
+    tp_factor = mesh_device.shape[1]
+    seq_per_chip = isl_per_chip
+    seq_len = isl_per_chip * sp_factor
+
+    signpost(f"embedding-{mesh_device.shape}-seq{seq_len}-v{vocab_size}-e{emb_dim}")
```
Comment (mbezuljTT) [inline models/demos/deepseek_v3_d_p/tests/pcc/test_parallel_embedding.py]: @copilot  change seq{seq_len} to isl_per_chip{isl_per_chip}; commit directly fix

File: models/demos/deepseek_v3_d_p/tests/pcc/test_parallel_embedding.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        ),
+    ],
+    indirect=["mesh_device", "device_params"],
+)
+def test_parallel_embedding(mesh_device, isl_per_chip, vocab_size, emb_dim):
+    """Test TtParallelEmbedding against torch reference."""
+
+    sp_factor = mesh_device.shape[0]
+    tp_factor = mesh_device.shape[1]
+    seq_per_chip = isl_per_chip
```
Comment (mbezuljTT) [inline models/demos/deepseek_v3_d_p/tests/pcc/test_parallel_embedding.py]: @copilot  fix 
remove     seq_per_chip = isl_per_chip and use isl_per_chip everywhere

**Review by ipotkonjak-tt — APPROVED** [review]

File: models/demos/deepseek_v3_d_p/tests/pcc/test_parallel_embedding.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    logger.debug(
+        f"Config: {mesh_device.shape=}, {sp_factor=}, {tp_factor=}, " f"{isl_per_chip=}, {vocab_size=}, {emb_dim=}"
+    )
+
+    assert (
+        isl_per_chip % ttnn.TILE_SIZE == 0
+    ), f"isl_per_chip ({isl_per_chip}) must be a multiple of TILE_SIZE ({ttnn.TILE_SIZE})"
+    assert emb_dim % tp_factor == 0, f"emb_dim ({emb_dim}) must be divisible by tp_factor ({tp_factor})"
+
+    ttnn.visualize_mesh_device(mesh_device)
```
Comment (pavlepopovic) [inline models/demos/deepseek_v3_d_p/tests/pcc/test_parallel_embedding.py:67]: remove?

**Review by pavlepopovic — APPROVED** [review]

File: models/demos/deepseek_v3_d_p/tests/pcc/test_parallel_embedding.py (hunk 3 above)
Comment (mbezuljTT) [reply models/demos/deepseek_v3_d_p/tests/pcc/test_parallel_embedding.py:67]: I find it useful when browsing long logs
